addons/account/views/base_document_layout_views.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="view_base_document_layout" model="ir.ui.view">
+            <field name="name">Document Layout</field>
+            <field name="model">base.document.layout</field>
+            <field name="inherit_id" ref="web.view_base_document_layout"/>
+            <field name="mode">primary</field>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='report_footer']" position="after">
+                    <field name="from_invoice" invisible="1"/>
+                    <field name="qr_code" string="QR Code" invisible="not from_invoice"/>
+                    <field name="acc_number" string="Iban" invisible="not qr_code" required="qr_code" placeholder="BE71096123456769"/>
```

Comment:
I'm not sure but I think it should be `IBAN`
```suggestion
                    <field name="acc_number" string="IBAN" invisible="not qr_code" required="qr_code" placeholder="BE71096123456769"/>
```

Btw, I don't know for the functional spec but according to [wikipedia](https://fr.wikipedia.org/wiki/International_Bank_Account_Number), IBAN is not used everywhere so maybe we should keep account number or bank account number?

Reply:
The functional need is to setup SEPA QR-codes quickly and afaik only IBAN bank accounts are allowed in that case, so there's nothing wrong about putting IBAN in the name of the field.